### DISCUSSION TITLE
Accessing a molecule property requires iterating over all properties (worst case)

### DISCUSSION THREAD

**Discussion by PatrickPenner** [rdkit/rdkit#7877] [Development]
I just came across this line: https://github.com/rdkit/rdkit/blob/a4fe959523fcf87108b157afc0327583e11e6b9d/Code/RDGeneral/Dict.h#L47

It defines the datatype of the Dict used for molecule properties as a std::vector of a pair struct that wraps properties. Is there a particular reason this is a std::vector instead of a kind of map?

I wasn't able to quickly find one in the discussions or the mailing list.

**Comment by bp-kelley** [comment]
Generally the size of the properties is small and scanning the vectors is faster than using a map.

**Reply by bp-kelley** [reply]
We switched from maps a long time ago, benchmarking is in the pull request

https://github.com/rdkit/rdkit/pull/896

**Reply by bp-kelley** [reply]
It also has a smaller memory footprint

**Reply by PatrickPenner** [reply]
Thanks. It looked a bit unintuitive. I assume there are enough property accesses in the molecule creation to consider those covered in the benchmark?

**Comment by bp-kelley** [comment]
Yes, properties are heavily used and tested IMO
